.github/workflows/scripts/github.mjs

```diff
@@ -0,0 +1,82 @@
+/**
+ * Generates a URL to the "Artifacts" section of a Github action run
+ *
+ * Unfortunately the best we can do here is a link to the "Artifacts" section as
+ * [the upload-artifact action doesn't provide
+ * the public URL](https://github.com/actions/upload-artifact/issues/50) :'(
+ *
+ * @param {number} runId - - The ID of the Github action run
+ * @returns {string} The URL to the "Artifacts" section of the given run
+ */
+export function githubActionArtifactsUrl (runId) {
+  return `https://github.com/alphagov/govuk-frontend/actions/runs/${runId}/#artifacts`
+}
+
+/**
+ * @param {Pick<GithubActionContext, "github" | "context" | "issueNumber">} githubContext - Github Action context
+ * @param {string} markerText - A unique marker used to identify the correct comment
+ * @param {string} bodyText - The body of the comment
+ */
+export async function comment (
+  { github, context, issueNumber },
+  markerText,
+  bodyText
+) {
+  const marker = `<!-- ${markerText} -->\n`
+  const body = `${marker}${bodyText}`
+
+  let commentId
+
+  /**
+   * @satisfies {IssueCommentsListParams}
+   */
+  const issueParameters = {
+    issue_number: issueNumber,
+    owner: context.repo.owner,
+    repo: context.repo.repo
+  }
+
+  /**
+   * Finds the first issue comment for which the `matcher` function returns `true`
+   *
+   * {@link https://github.com/peter-evans/find-comment/blob/main/src/find.ts}
+   */
+  for await (const { data: comments } of github.paginate.iterator(
+    github.rest.issues.listComments,
+    issueParameters
+  )) {
+    const comment = comments.find((comment) => !!comment.body?.includes(marker))
+
+    if (comment) {
+      commentId = comment.id
+    }
+  }
+
+  /**
+   * Create GitHub issue comment
+   *
+   * Updates existing commentby ID if available
+   */
+  await (!commentId
+    ? github.rest.issues.createComment({ ...issueParameters, body })
+    : github.rest.issues.updateComment({
+      ...issueParameters,
+      body,
+      comment_id: commentId
+    }))
+}
+
+/**
+ * @typedef {object} GithubActionContext
```

Comment:
This is an attempt at formalising the structure of functions we'd call from Github action's script into something of the shape of `function (githubActionContext, ...argumentsSpecificToTheFunction)`.

The idea is to have a one stop object to collect all the values from the github context in the action via `${{github.<xyz>}}` and clearly separate them from the parameters that are actually specific to the function's job.

This is only a starting point to set a direction, and can be adjusted as we work more with Github action. If we realise it needs splitting into different things, or smaller types or whatever, it can happily be refactored then.

Reply:
Sounds good

Mind if we narrow down `GitHubActionContext` to only what we get from GitHub?

Good to separate out our own params (either as actual params or an object)

```suggestion
 * @typedef {object} GitHubActionContext
```

Reply:
Argh sorry, wasn't clear was I? 🤦‍♂️ 

I mean the objects and tooling we get from [`actions/github-script`](https://github.com/actions/github-script/tree/v6/#readme)

>* `github` A pre-authenticated [octokit/rest.js](https://octokit.github.io/rest.js) client with pagination plugins
context An object containing the [context of the workflow run](https://github.com/actions/toolkit/blob/main/packages/github/src/context.ts)
>* `core` A reference to the [@actions/core](https://github.com/actions/toolkit/tree/main/packages/core) package
>* `glob` A reference to the [@actions/glob](https://github.com/actions/toolkit/tree/main/packages/glob) package
>* `io` A reference to the [@actions/io](https://github.com/actions/toolkit/tree/main/packages/io) package
>* `exec` A reference to the [@actions/exec](https://github.com/actions/toolkit/tree/main/packages/exec) package
>* `fetch` A reference to the [node-fetch](https://github.com/node-fetch/node-fetch) package

I liked this approach:

```js
export async function comment ({ github, context }, issueNumber, markerText, bodyText) {}
```

Unlike the shared `github` and `context` objects we're likely to vary these per function:

```
issueNumber
markerText
bodyText
```